service/models.py

```diff
@@ -84,4 +93,34 @@ def init_db(cls, app):
         # This is where we initialize SQLAlchemy from the Flask app
         db.init_app(app)
         app.app_context().push()
-        db.create_all()  # make our sqlalchemy tables
\ No newline at end of file
+        db.create_all()  # make our sqlalchemy tables
+    
+
+    @classmethod
+    def all(cls):
+        """ Returns all of the Recommendation in the database """
+        logger.info("Processing all Recommendations")
+        return cls.query.all()
+
+
+    @classmethod
+    def find_by_recommendation_type(cls, recommendation_type):
+        """Returns all recommendations with the given recommendation_type
+
+        Args:
+            recommendation_type (string): the recommendation_type of the Recommendations you want to match
+        """
+        logger.info("Processing recommendation_type query for %s ...",
+                    recommendation_type)
+        return cls.query.filter(cls.recommendation_type == recommendation_type)
```

Comment:
I believe this should be cls.relationship though I am not exactly sure. This branch does not compile when I am running nosetests locally. Actually I think you could remove this chunk of code and we could say we support list all recommendations.